keps/sig-node/xxx-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1408 @@
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+* More flexibility:
+  * Arbitrary, resource-specific setup and cleanup actions
+  * Over-the-fabric resources
+  * Custom matching of resource requests with available resources,
+    including handling of optional resource requests
+* User-friendly API for describing resource requests
+* Allow resource management plugins that can be developed and deployed
+  separately from Kubernetes and are independent of specific container
+  runtimes.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* An extension of the model that kube-scheduler has about
+  resources. Instead, it will need information from the plugin for
+  each resource request to determine where a Pod using the resource
+  might run. The [Representing Compute Resources in Kubernetes
+  proposal](https://docs.google.com/document/d/1666PPUs4Lz56TqKygcy6mXkNazde-vwA7q4e5H92sUc/edit#)
+  had some ideas what information the scheduler might need (“supports
+  overcommit”, “fractional”), but ultimately any choice regarding that
+  will only work for certain kinds of resources.
+* A standardized way of describing available resources. Only allocated
+  resources are visible through the APIs defined below. How to
+  describe available resources is plugin specific because it depends
+  on the kind of resource which attributes might be relevant. Plugins
+  should use and document their individual approach for this (for
+  example, defining a CRD and publishing through that).
+* An abstraction layer for resource requests, i.e., something like a
+  “I want some kind of GPU”. Users will need to know about specific
+  resource plugins and which parameters they support. Portability of
+  workloads could be added on top of this proposal by introducing the
+  selection of a resource implementation through labels and
+  standardizing those labels and the associated parameters. The
+  [Resource Class
+  Proposal](https://docs.google.com/document/d/1qKiIVs9AMh2Ua5thhtvWqOqW0MSle_RV3lfriO1Aj6U/edit#heading=h.jzfmfdca34kj)
+  included such an approach.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+The proposal is that a plugin handles all operations that are specific
+to the resources managed by that plugin. This includes operations at
+the control plane level (tracking where in the cluster resources are
+available, helping with pod scheduling decisions, allocating resources
+when requested) as well as the node level (preparing container
```

Comment:
"allocating resources when requested"

This is going to be a tough one to sell. It has a long held opinion that the kubelet itself should always have the final say in any resource allocation decisions. Its decisions can be (and even are today) influenced by the plugins, but the kubelet always makes the final decision.

Reply:
kubelet can have the final say about resources that it manages.

But in this KEP, kubelet is going to be oblivious of the resources managed by a plugin. All that kubelet needs to do (can can do) is reconfiguring the container runtime to expose resources.